code/datums/achievements/_achievement_data.dm

```diff
@@ -20,33 +18,36 @@
 	initialized = TRUE
 	load_all_achievements() //So we know which achievements we have unlocked so far.
 
-	var/datum/asset/spritesheet/simple/assets = get_asset_datum(/datum/asset/spritesheet/simple/achievements)
-	AchievementIcons = list()
-	for(var/achievement_type in SSachievements.achievements)
-		var/datum/award/achievement = SSachievements.achievements[achievement_type]
-		var/list/SL = list()
-		SL["htmltag"] = assets.icon_tag(achievement.icon)
-		AchievementIcons[achievement.name] += list(SL)
-
-///Saves any out-of-date achievements to the hub.
+///Saves any out-of-date achievements to db.
 /datum/achievement_data/proc/save()
 	for(var/T in data)
 		var/datum/award/A = SSachievements.awards[T]
-
-		if(data[T] != original_cached_data[T])//If our data from before is not the same as now, save it to the hub. This check prevents unnecesary polling.
+		if(data[T] != original_cached_data[T])//If our data from before is not the same as now, save it to db.
 			A.save(key,data[T])
 
-///Loads data for all achievements to the caches.
-/datum/achievement_data/proc/load_all()
-	for(var/T in subtypesof(/datum/award))
-		get_data(T)
-
 /datum/achievement_data/proc/load_all_achievements()
 	set waitfor = FALSE
-	for(var/T in subtypesof(/datum/award/achievement))
-		get_data(T)
+	
+	var/list/kv = list()
+	var/datum/DBQuery/Query = SSdbcore.NewQuery("SELECT achievement_key,value FROM [format_table_name("achievements")] WHERE ckey = '[sanitizeSQL(key)]'")
+	if(!Query.Execute())
+		qdel(Query)
+		return
+	while(Query.NextRow())
+		var/key = Query.item[1]
```

Comment:
I can't comment on the line but line 4 `var/key` is used to store a ckey and this is confusing when that's an actual client var as well, rename it to something like `var/owner_ckey`